homeassistant/components/fastdotcom/manifest.json

```diff
@@ -6,5 +6,6 @@
   "documentation": "https://www.home-assistant.io/integrations/fastdotcom",
   "iot_class": "cloud_polling",
   "loggers": ["fastdotcom"],
+  "quality_scale": "gold",
```

Comment:
- A diagnostics platform is missing.
- Please post the coverage result from a local pytest test run of the integration.
- Some check boxes are not filled in which are already coveredvia use of coordinator, like not available due to internet outtage.

Reply:
Adding a diagnostics platform should be done in a separate PR.

Reply:
Test coverage result:

```
Test session starts (platform: linux, Python 3.11.4, pytest 8.0.2, pytest-sugar 1.0.0)
cachedir: .pytest_cache
rootdir: /workspaces/ha-core
configfile: pyproject.toml
plugins: timeout-2.2.0, socket-0.7.0, sugar-1.0.0, cov-4.1.0, aiohttp-1.0.5, syrupy-4.6.1, respx-0.20.2, test-groups-1.0.3, picked-0.5.0, pytest_freezer-0.4.8, unordered-0.5.2, asyncio-0.23.5, xdist-3.3.1, requests-mock-1.11.0, anyio-4.1.0
asyncio: mode=Mode.AUTO
collected 14 items                                                                                                                                                                          

 tests/components/fastdotcom/test_config_flow.py::test_user_form ✓                                                                                                              7% ▊         
 tests/components/fastdotcom/test_config_flow.py::test_single_instance_allowed[user] ✓                                                                                         14% █▌        
 tests/components/fastdotcom/test_config_flow.py::test_single_instance_allowed[import] ✓                                                                                       21% ██▎       
 tests/components/fastdotcom/test_config_flow.py::test_import_flow_success ✓                                                                                                   29% ██▉       
 tests/components/fastdotcom/test_coordinator.py::test_fastdotcom_data_update_coordinator ✓                                                                                    36% ███▋      
 tests/components/fastdotcom/test_diagnostics.py::test_get_config_entry_diagnostics ✓                                                                                          43% ████▍     
 tests/components/fastdotcom/test_init.py::test_unload_entry ✓                                                                                                                 50% █████     
 tests/components/fastdotcom/test_init.py::test_from_import ✓                                                                                                                  57% █████▊    
 tests/components/fastdotcom/test_init.py::test_delayed_speedtest_during_startup ✓                                                                                             64% ██████▌   
 tests/components/fastdotcom/test_init.py::test_service_deprecated ✓                                                                                                           71% ███████▎  
 tests/components/fastdotcom/test_sensor.py::test_fastdotcom_data_update_coordinator ✓                                                                                         79% ███████▉  
 tests/components/fastdotcom/test_service.py::test_service ✓                                                                                                                   86% ████████▋ 
 tests/components/fastdotcom/test_service.py::test_service_unloaded_entry ✓                                                                                                    93% █████████▍
 tests/components/fastdotcom/test_service.py::test_service_removed_entry ✓                                                                                                    100% ██████████

---------- coverage: platform linux, python 3.11.4-final-0 -----------
Name                                                 Stmts   Miss  Cover   Missing
----------------------------------------------------------------------------------
homeassistant/components/fastdotcom/__init__.py         31      0   100%
homeassistant/components/fastdotcom/config_flow.py      18      0   100%
homeassistant/components/fastdotcom/const.py            10      0   100%
homeassistant/components/fastdotcom/coordinator.py      14      0   100%
homeassistant/components/fastdotcom/diagnostics.py      10      0   100%
homeassistant/components/fastdotcom/sensor.py           28      0   100%
homeassistant/components/fastdotcom/services.py         24      0   100%
----------------------------------------------------------------------------------
TOTAL                                                  135      0   100%

---------------------------------------------------------------------------------- snapshot report summary ----------------------------------------------------------------------------------
1 snapshot passed.

Results (2.84s):
      14 passed
```

Reply:
PR #111525 created for the diagnostics platform.